zokrates_core/src/absy/variable.rs

```diff
@@ -2,37 +2,39 @@ use crate::absy::Node;
 use crate::types::Type;
 use std::fmt;
 
-#[derive(Serialize, Deserialize, Clone, PartialEq, Hash, Eq)]
-pub struct Variable {
-    pub id: String,
+use crate::absy::Identifier;
+
+#[derive(Clone, PartialEq, Hash, Eq)]
+pub struct Variable<'ast> {
+    pub id: Identifier<'ast>,
```

Comment:
Variables do not contain owned strings anymore, but references to the source string